watchmal/dataset/cnn_mpmt/transformations.py

```diff
@@ -14,4 +16,50 @@ def horizontal_flip(data):
     return flip(data[horizontal_flip_mpmt_map, :, :, ], [2])
 
 def vertical_flip(data):
-    return flip(data[vertical_flip_mpmt_map, :, :], [1])
\ No newline at end of file
+    return flip(data[vertical_flip_mpmt_map, :, :], [1])
+
+
+def rotation180(data, barrel_rows):
+    transform_data = data.clone()
+    transform_data[:, barrel_rows, :] = torch.tensor(np.roll(transform_data[:, barrel_rows, :], 20, 2))
+    
+    l_index = data.shape[2]/2 - 1
+    r_index = data.shape[2]/2
+
+    l_endcap_ind = int(l_index - 4)
+    r_endcap_ind = int(r_index + 4)
+    
+    top_end_cap = data.clone()[:, barrel_rows[-1]+1:, l_endcap_ind:r_endcap_ind+1]
+    bot_end_cap = data.clone()[:, :barrel_rows[0], l_endcap_ind:r_endcap_ind+1]
+    
+    vhflip_top = horizontal_flip(vertical_flip(top_end_cap))
+    vhlfip_bot = horizontal_flip(vertical_flip(bot_end_cap))
+    
+    transform_data[:, barrel_rows[-1]+1:, l_endcap_ind : r_endcap_ind+1] = vhflip_top
+    transform_data[:, :barrel_rows[0], l_endcap_ind : r_endcap_ind+1] = vhlfip_bot
+    
+    return transform_data
+
+
+def mpmtPadding(data, barrel_rows):
+    half_len_index = int(data.shape[2]/2)
+    horiz_pad_data = torch.cat((data, torch.zeros_like(data[:, :, :half_len_index])), 2)
+    horiz_pad_data[:, :, 2*half_len_index:] = torch.tensor(0, dtype=torch.float64)
+    horiz_pad_data[:, barrel_rows, 2*half_len_index:] = data[:, barrel_rows, :half_len_index]
+    
+    l_index = data.shape[2]/2 - 1
+    r_index = data.shape[2]/2
+    
+    l_endcap_ind = int(l_index - 4)
+    r_endcap_ind = int(r_index + 4)
+    
+    top_end_cap = data[:, barrel_rows[-1]+1:, l_endcap_ind:r_endcap_ind+1]
+    bot_end_cap = data[:, :barrel_rows[0], l_endcap_ind:r_endcap_ind+1]
+    
+    vhflip_top = horizontal_flip(vertical_flip(top_end_cap))
+    vhflip_bot = horizontal_flip(vertical_flip(bot_end_cap))
+    
+    horiz_pad_data[:, barrel_rows[-1]+1:, l_endcap_ind + int(data.shape[2]/2) : r_endcap_ind + int(data.shape[2]/2) + 1] = vhflip_top
+    horiz_pad_data[:, :barrel_rows[0], l_endcap_ind + int(data.shape[2]/2) : r_endcap_ind + int(data.shape[2]/2) + 1] = vhflip_bot
+    
+    return horiz_pad_data
```

Comment:
Now that the flips and rotation180 are in the CNNmPMTDataset object instead, I think they can be removed from here.
And then the mpmtPadding should also be moved to the dataset object where the new flip and rotation180 code is.